src/api/extension/wallet.js

```diff
@@ -496,3 +496,155 @@ export const withdrawalTx = async (account, delegation, protocolParameters) => {
+
+  certificates.add(
+    Loader.Cardano.Certificate.new_stake_deregistration(
+      Loader.Cardano.StakeDeregistration.new(
+        Loader.Cardano.StakeCredential.from_keyhash(
+          Loader.Cardano.Ed25519KeyHash.from_bytes(
+            Buffer.from(account.stakeKeyHash, 'hex')
+          )
+        )
+      )
+    )
+  );
+
+  txBuilder.set_certs(certificates);
+
+  const change = selection.change;
+  const changeMultiAssets = change.multiasset();
+
+  // check if change value is too big for single output
+  if (
+    changeMultiAssets &&
+    change.to_bytes().length * 2 > protocolParameters.maxValSize
+  ) {
+    const partialChange = Loader.Cardano.Value.new(
+      Loader.Cardano.BigNum.from_str('0')
+    );
+
+    const partialMultiAssets = Loader.Cardano.MultiAsset.new();
+    const policies = changeMultiAssets.keys();
+    const makeSplit = () => {
+      for (let j = 0; j < changeMultiAssets.len(); j++) {
+        const policy = policies.get(j);
+        const policyAssets = changeMultiAssets.get(policy);
+        const assetNames = policyAssets.keys();
+        const assets = Loader.Cardano.Assets.new();
+        for (let k = 0; k < assetNames.len(); k++) {
+          const policyAsset = assetNames.get(k);
+          const quantity = policyAssets.get(policyAsset);
+          assets.insert(policyAsset, quantity);
+          //check size
+          const checkMultiAssets = Loader.Cardano.MultiAsset.from_bytes(
+            partialMultiAssets.to_bytes()
+          );
+          checkMultiAssets.insert(policy, assets);
+          const checkValue = Loader.Cardano.Value.new(
+            Loader.Cardano.BigNum.from_str('0')
+          );
+          checkValue.set_multiasset(checkMultiAssets);
+          if (
+            checkValue.to_bytes().length * 2 >=
+            protocolParameters.maxValSize
+          ) {
+            partialMultiAssets.insert(policy, assets);
+            return;
+          }
+        }
+        partialMultiAssets.insert(policy, assets);
+      }
+    };
+    makeSplit();
+    partialChange.set_multiasset(partialMultiAssets);
+    const minAda = Loader.Cardano.min_ada_required(
+      partialChange,
+      false,
+      Loader.Cardano.BigNum.from_str(protocolParameters.coinsPerUtxoWord)
+    );
+    partialChange.set_coin(minAda);
+
+    txBuilder.add_output(
+      Loader.Cardano.TransactionOutput.new(
+        Loader.Cardano.Address.from_bech32(account.paymentAddr),
+        partialChange
+      )
+    );
+  }
+
+  txBuilder.set_ttl(protocolParameters.slot + TX.invalid_hereafter);
+  txBuilder.add_change_if_needed(
+    Loader.Cardano.Address.from_bech32(account.paymentAddr)
+  );
+
+  const transaction = Loader.Cardano.Transaction.new(
+    txBuilder.build(),
+    Loader.Cardano.TransactionWitnessSet.new()
+  );
+
+  const size = transaction.to_bytes().length * 2;
+  if (size > protocolParameters.maxTxSize) throw ERROR.txTooBig;
+
+  return transaction;
+};
```

Comment:
Could this be unit-tested? (or perhaps could we have an issue logged to unit-test this? Me or someone else could take it)

Reply:
Yes this could be unit tested on the current branch or after the merge. If you are interested in giving it a try that would be awesome.

Reply:
@pascallapointe I’d love to do that! If you don’t mind waiting for the weekend 😆 How can I reach out to you? (what Discord? — I’m guessing Discord is the place to go)

Reply:
I prefer telegram, you'll find me joining the berry channel (https://t.me/berry_pool). I'm having a hard time following all activities on Discord ..!